source/Calamari.Common/Plumbing/Extensions/LogExtensions.cs

```diff
@@ -23,5 +23,16 @@ public static void LogMetric(this ILog log, string metricName, object metricValu
 
             log.WriteServiceMessage(new ServiceMessage(ServiceMessageNames.CalamariDeploymentMetric.Name, serviceMessageParameters));
         }
+
+        public static void AddValueToRedact(this ILog log, string? value, string replacement)
+        {
+            if (!(log is RedactedValuesLogger redactedValuesLogger))
```

Comment:
How will the consumer of this method know if their logger is able to redact?

Reply:
They wont... I was thinking this exception is more for developers as if they're hitting this issue, something hasn't been setup correctly. I don't expect anyone to hit this exception if they have just injected `ILog` like normal.

Reply:
Sorry, I was unclear - by consumer, I meant the developers.

Reply:
Yeah I understand, it's not ideal... the problem is that even if I register `RedactedValuesLogger` as itself so that when redacted values need to be added, the class can just inject the redacted logger directly (rather than `ILog`, that doesn't guarantee that the same logger will be used when we log stuff. This would mean that data we want redacted could be logged as-is without us knowing (unless we manually check). I went this way because I thought that an exception when trying to add redacted values is the better result.

Unsure if there is a better solution... 

CC: @zentron

Reply:
Solved this issue after discussion with @MissedTheMark by moving the RedactedValuesLogger functionality into the AbstractLog and I even wrote a few unit tests around the Redacting of values.

I updated the ILog interface to have the ability to add redacted values as well.

This feels so much cleaner as now we're guaranteed to redact values if they are give to the logger no matter where the logger is injected/passed etc.